torch/nn/modules/rnn.py

```diff
@@ -429,6 +436,16 @@ def forward(self, input, hx=None):  # noqa: F811
             max_batch_size = int(batch_sizes[0])
         else:
             batch_sizes = None
+            is_batched = input.dim() == 3
+            batch_dim = 0 if self.batch_first else 1
+            if not is_batched:
+                input = input.unsqueeze(batch_dim)
+                if hx is not None:
+                    assert hx.dim() == 2
```

Comment:
I think raising a `RuntimeError` is preferred to asserting, and I like the error messages you created for the `GRU` case - can we get similar ones here as well?